src/components/autocomplete/js/autocompleteController.js

```diff
@@ -466,12 +466,20 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        if (!shouldProcessEscape()) return;
         event.stopPropagation();
         event.preventDefault();
-        if ($scope.searchText) clearValue();
 
-        // Force the component to blur if they hit escape
-        doBlur(true);
+        clearValue(shouldClearOnEscape() && $scope.searchText);
+
+        if (shouldBlurOnEscape()) {
+          // Force the component to blur if they hit escape
+          doBlur(true);
+        }
+        else {
```

Comment:
Nitpicky: This `else` statement should be next to the brace (consistency) to follow the Google Code Style Conventions